### ISSUE TITLE
Marking styles (e.g. superscript) in a more appealing way in the markup itself too

### ISSUE THREAD

**Issue by wolftune** [ChordPro/chordpro#13]
in NNS, 5 with superscript 7 is clear, but looking at the plain text source file, [57] is odd looking. I originally thought maybe [5^7] would be good, but I want to reserve ^ for something else. Not sure what's the best solution.

**Comment by sciurius** [comment]
Formatting printed chords in a more elegant way is nice to have, but currently too far from the goals of ChordPro. There are many strings attached that need to be sorted out first.
We'll keep this as a feature request for some future release.

**Comment by rustianimal** [comment]
This is an incredibly powerful thing to have. In Graphic User Interfaces, designing a system to work effectively for the eyes is extremely important. When we read, we do not read the individual characters in a word, but the outline shape of the word. This is critical to speed reading where people are specifically taught to ignore vowels and many other characters and focus on the shape of words. In fact it is possible to take out pretty much all vowels and quite a lot of other characters  and still make perfect sense of the sentence. 

To prove the point, convert the whole of the above paragraph to capitals and then try to read it. This is exactly what the chord shapes in the current version of ChordPro are like. If you have any form of reading difficultly (as I have), it is almost impossible to distinguish the chords from the lyric, even when using colour. What makes an absolutely MASSIVE difference using superscript and subscript in the chord description as has been the case on sheet music for decades.

It is time that ChordPro got itself up to date and became supportive of those with visual and learning impairments by putting the tools in place to support them. I would propose the not just the options described above, but to also allow substations such as ^4 instead of 'sus' and to subscript 'm' and 'M' for minor or major chords. 

Supporting standard html methods for this would enable would open up the versatility of the ChordPro standard to allow custom chord conversions to assist those like me who NEED this kind of formatting to be able to play in a live context. Without this functionality, there is no option but for me to stick with traditional paper and ink.

**Comment by sciurius** [comment]
This has been moved to the [forum](https://groups.io/g/ChordPro/topic/rfd_formatting_of_chords/71726073) for further discussion.

**Comment by sciurius** [comment]
@rustianimal May I urge you to participate in the [forum](https://groups.io/g/ChordPro/topic/rfd_formatting_of_chords/71726073) discussion?

**Comment by sciurius** [comment]
Recent versions support defining display forms for chords, including Pango markup. See e.g. https://github.com/ChordPro/chordpro/blob/master/lib/App/Music/ChordPro/res/config/brandtroemer.json.

Does this solve the problem?

**Comment by wolftune** [comment]
@sciurius that looks very promising. But the example shows every single chord in a pattern spelled out. That's tedious and has room for human error. If I want to change how m7 chords look, I have to make sure not to miss every instance in the list. It would be far better if the style could parse the parts of chord names.

I'm going to update this issue actually since I don't want to use the ^ specifically. I want that available for another thing I will open in a new issue.

**Comment by lepaule** [comment]
After playing around with the "display" tag in config file "chords" directive, I found one little problem. If the chord is understood by ChordPro (e.g. Edim7), the PDF output will not format the chord text in the lyrics section. The chord text will format correctly above the diagram.
If the chord is not understood (e.g. Fbdim7), it is formatted as per "display" in both the lyric and diagram sections.
I have tried multiple workarounds and the only one that works is to format the name of the chord something ChordPro doesn't understand.
I have attached a pdf demonstrating what I mean.
[ChordPro display issue.pdf](https://github.com/ChordPro/chordpro/files/8274229/ChordPro.display.issue.pdf)
I have also attached an Excel file which builds the json string for @wolftune to play around with.
[ChordPro json chord builder.xlsx](https://github.com/ChordPro/chordpro/files/8274242/ChordPro.json.chord.builder.xlsx)

**Comment by sciurius** [comment]
@lepaule can you include the .cho and config?

**Comment by lepaule** [comment]
@sciurius, sure. I put the font files I'm using in too.
[lepaule ChordPro Configs.zip](https://github.com/ChordPro/chordpro/files/8275737/lepaule.ChordPro.Configs.zip)

**Comment by lepaule** [comment]
@sciurius, here's a couple other examples. I was motivated by including a subscript for chord inversions like Where Is My Mind. But then in something like What A Wonderful World and after adding Brandt-Roemer chord formatting, some of the chords displayed one way and some not. In these example files I have fixed the formatting issue by making the chords unrecognized by ChordPro.
[other examples.zip](https://github.com/ChordPro/chordpro/files/8277288/other.examples.zip)
.

**Comment by sciurius** [comment]
A lot has changed in this area since 5.985 and AFAICS 5.987 produces correct results. See attachment.

5.987 requires Text::Layout version 0.029. You need to install/upgrade this as well.

BTW, to rise all chords a bit it is easier to adjust the `pdf.spacing.chords` setting than to add `<span rise='-200'>...</span>` to all chords.

[x.pdf](https://github.com/ChordPro/chordpro/files/8283531/x.pdf)

**Comment by lepaule** [comment]
Ha ha. Does ChordPro not check for updates? My computer's only a few months old, so I didn't think to check. I remember 15 years ago when I would always check for updates before posting to a forum. I will look into that and have a little more fun.
The rise attribute was because I had just found the Cadman font (only one I found with flat, natural and sharp characters in all four font files) and was playing around with proper placements of superscript and subscript text. Using Excel to build the json only takes a couple seconds to change every chord. There was no noticeable change in print time, but I will look into spacing directives. Thanks.
